activation/nipost_test.go

```diff
@@ -41,6 +48,337 @@ func defaultLayerClockMock(ctrl *gomock.Controller) *MocklayerClock {
 	return mclock
 }
 
+func Test_NIPost_PostClientHandling(t *testing.T) {
+	t.Run("connect then complete", func(t *testing.T) {
+		// post client connects, starts post, then completes successfully
+
+		sig, err := signing.NewEdSigner()
+		require.NoError(t, err)
+
+		ctrl := gomock.NewController(t)
+		poetDb := NewMockpoetDbAPI(ctrl)
+		mclock := defaultLayerClockMock(ctrl)
+		postService := NewMockpostService(ctrl)
+		postClient := NewMockPostClient(ctrl)
+
+		logger := zaptest.NewLogger(t)
+		nb, err := NewNIPostBuilder(
+			localsql.InMemory(),
+			poetDb,
+			postService,
+			[]types.PoetServer{},
+			logger,
+			sig,
+			PoetConfig{},
+			mclock,
+		)
+		require.NoError(t, err)
```

Comment:
Isn't there a bit too much repetition?
Maybe the code is more clear the way, but on the other hand maybe some kind of "fixture" ("tester") like in https://github.com/spacemeshos/go-spacemesh/blob/9c9f42fac2da76595430673ecc59df39b74d9466/fetch/fetch_test.go#L48 or https://github.com/spacemeshos/go-spacemesh/blob/9c9f42fac2da76595430673ecc59df39b74d9466/hare3/hare_test.go#L551 could help a little bit?